### PR TITLE
Admin dashboard fixes

### PR DESCRIPTION
**Pull request by expertdeveloper5821**
1> Admin Dashboard Users fixed and refactored.
2> Admin Dashboard Spectator fixed and refactored.
3> Admin Dashboard Rooms fixed and refactored.

### CODE REVIEW

File: src/Components/CommonComponent/SearchFilter.tsx [hunk 1]
```diff
@@ -1,21 +1,34 @@
...
+
+export function SearchFilter({ handleSearch,onChange }: any) {
+
+  const [searchFilter, setSearchFilter] = useState("");
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    onChange(e);
+    setSearchFilter(e?.target?.value);
+  };
+  console.log("searchFilter",searchFilter)
```
Comment (Ashok-tgc) [inline src/Components/CommonComponent/SearchFilter.tsx]: remove console.log

File: src/Components/CommonComponent/SearchFilter.tsx [hunk 2]
```diff
@@ -1,21 +1,34 @@
...
-            <Input placeholder="Date and Time" type="text"></Input>
-            <div className={styles.select_wrapper}></div>
-            <div className={styles.button_wrapper}>
-              <Button className={styles.searchbutton} varient="contained">
+      <div className={styles?.dashboard_content}>
+
+        <div className={styles?.content_wrapper}>
+          <div className={styles?.input_desc}>
+            <Input className={styles?.search_input_box} placeholder="Search" type="search" onChange={handleInputChange}></Input>
+            {/* <Input placeholder="Date and Time" type="text"></Input> */}
```
Comment (Ashok-tgc) [inline src/Components/CommonComponent/SearchFilter.tsx]: remove comments if not necessary and format code

File: src/Components/CommonComponent/SearchFilter.tsx [hunk 3]
```diff
@@ -25,3 +38,11 @@ export function SearchFilter() {
     </>
   );
 }
+
+// width: 100%;
```
Comment (Ashok-tgc) [inline src/Components/CommonComponent/SearchFilter.tsx]: remove comments

File: src/Components/CommonComponent/Table/Table.tsx [hunk 4]
```diff
@@ -22,133 +23,229 @@ export interface StudentProfile {
...
+        }
+      })
+      .join('');
   }
 
-  const handleEdit = (studentData: studentData) => {
-    console.log('Edit student data:', studentData);
+  const handleSort = (key: string, arrow:any) => {
+    const newKey = toCamelCase(key);
+      console.log("INSIDEEEE 0 key===>",key,"type",type,"newKey",newKey,"arrow",arrow,"sortedData",sortedData,"columns",columns);
```
Comment (Ashok-tgc) [inline src/Components/CommonComponent/Table/Table.tsx]: remove logs everywhere

File: src/Components/CommonComponent/Table/Table.tsx [hunk 5]
```diff
@@ -22,133 +23,229 @@ export interface StudentProfile {
...
+                    </TableCell>
+                  </TableRow>
+                );
+              }
+              else if(type === "USERS"){
+                console.log("USERS => STUDENTDATA AND INDEX AND SORTEDDATA AND TYPE 2.1==>",studentData,index,sortedData,type)
+
+                return (
+                  <TableRow className={styles?.table_rowdata} key={index}>
+                    <TableCell className={styles?.table_cell}>{studentData?.fullName}</TableCell>
```
Comment (Ashok-tgc) [inline src/Components/CommonComponent/Table/Table.tsx]: is it possible to reuse a single table component for users and spectators?

File: src/app/adminDashboard/room/page.tsx [hunk 6]
```diff
@@ -5,95 +5,135 @@ import React, { useState, useEffect } from 'react';
...
+    setRoomData(filteredResults);
+  };
 
   return (
     <>
-        <div className={styles.main_container} id="mainLayoutContainerInner">
-          <div className={styles.abcd}>
-            <div className={styles.sidebar_wrapper}>
+        <div className={styles?.main_container} id="mainLayoutContainerInner">
+          <div className={styles?.abcd}>
```
Comment (Ashok-tgc) [inline src/app/adminDashboard/room/page.tsx]: why "?" in className?

File: src/app/adminDashboard/users/page.tsx [hunk 7]
```diff
@@ -1,7 +1,120 @@
...
+  const columns: string[] = [
+    'Full Name',
+    'User Name',
+    'Email',
+  ];
+
+  const fetchTournaments = async (searchVal:string) => {
+    try {
+      const token = localStorage.getItem('jwtToken');
+      const response = await sendRequest(`/user/getalluser?search=${searchVal}`, {
```
Comment (Ashok-tgc) [inline src/app/adminDashboard/users/page.tsx]: is it possible to a create service for all these apis and use that service here? please check services folder

File: src/Components/CommonComponent/SearchFilter.tsx (hunk 1 above)
Comment (expertdeveloper5821) [reply src/Components/CommonComponent/SearchFilter.tsx]: resolved

File: src/Components/CommonComponent/SearchFilter.tsx (hunk 2 above)
Comment (expertdeveloper5821) [reply src/Components/CommonComponent/SearchFilter.tsx]: resolved

File: src/Components/CommonComponent/SearchFilter.tsx (hunk 3 above)
Comment (expertdeveloper5821) [reply src/Components/CommonComponent/SearchFilter.tsx]: resolved

File: src/Components/CommonComponent/Table/Table.tsx (hunk 4 above)
Comment (expertdeveloper5821) [reply src/Components/CommonComponent/Table/Table.tsx]: resolved

File: src/Components/CommonComponent/Table/Table.tsx (hunk 5 above)
Comment (expertdeveloper5821) [reply src/Components/CommonComponent/Table/Table.tsx]: resolved

File: src/app/adminDashboard/room/page.tsx (hunk 6 above)
Comment (expertdeveloper5821) [reply src/app/adminDashboard/room/page.tsx]: resolved

File: src/app/adminDashboard/users/page.tsx (hunk 7 above)
Comment (expertdeveloper5821) [reply src/app/adminDashboard/users/page.tsx]: resolved

File: src/app/adminDashboard/users/page.tsx [hunk 8]
```diff
@@ -6,43 +6,30 @@ import TableData from '@/Components/CommonComponent/Table/Table';
...
+import { deleteRoleService, getAllFilteredUsersListService } from '@/services/authServices';
+import { SpectatorDataType } from '@/types/spectatorTypes';
 
 function page() {
-  const [ wholeUserData, setWholeUserData] = useState<SpectatorDataType[] | []>([]);
-  const [userData, setUserData] = useState<SpectatorDataType[] | []>([]);;
+  const [wholeUserData, setWholeUserData] = useState<SpectatorDataType[] | []>([]);
+  const [userData, setUserData] = useState<SpectatorDataType[] | []>([]);
   const [isLoading, setIsLoading] = useState<boolean>(true);
   const imageIcon: string = 'user';
```
Comment (Ashok-tgc) [inline src/app/adminDashboard/users/page.tsx]: remove unused vars

File: src/app/adminDashboard/users/page.tsx [hunk 9]
```diff
@@ -72,10 +56,11 @@ function page() {
 
   const handleSearch = (e) => {
     const { name, value } = e.target;
```
Comment (Ashok-tgc) [inline src/app/adminDashboard/users/page.tsx]: name also not used, remove all unused vars

File: src/services/authServices.ts [hunk 10]
```diff
@@ -1,6 +1,9 @@
 import { sendRequest } from '@/utils/axiosInstanse';
 import { serviceUrls } from './serviceUrls';
 import { SignupFormValuesType } from '@/Components/pageComponents/auth/authInterfaces';
+import { RegisterSpectatorValuesType, updateRoleValuesType } from '@/types/spectatorTypes';
```
Comment (Ashok-tgc) [inline src/services/authServices.ts:4]: move to new file with different name, authService is for auth apis

File: src/app/adminDashboard/spectator/page.tsx [hunk 11]
```diff
@@ -219,84 +290,124 @@ export default function Modal() {
...
+                deleteroom={deleteroom}
+                handleEdit={handleEdit}
               />
             )}
           </div>
         </div>
       </div>
 
-      {modal && (
+      {modal?.isOpen && (
```
Comment (Ashok-tgc) [inline src/app/adminDashboard/spectator/page.tsx:227]: move modal to new component

File: src/app/adminDashboard/spectator/page.tsx [hunk 12]
```diff
@@ -1,199 +1,269 @@
...
-    if (!getSpectatorId) {
-      const initialFormData: FormCreate = {
-        fullName: '',
-        userName: '',
-        email: '',
-        password: '',
-        role: specRoleId,
-      };
-      setFormData(initialFormData);
+  const addFormValidations = (name, value) => {
```
Comment (Ashok-tgc) [inline src/app/adminDashboard/spectator/page.tsx]: refactor this function

File: src/app/adminDashboard/room/page.tsx [hunk 13]
```diff
@@ -1,103 +1,120 @@
...
-    'Game Type',
-    'Map Type',
-    'Version',
-    'Time',
-    'Date',
-  ];
+  const fetchTournaments = async (searchVal: string) => {
+    try {
+      const token = localStorage.getItem('jwtToken');
+      const response = await sendRequest(`/room/rooms?search=${searchVal}`, {
```
Comment (Ashok-tgc) [inline src/app/adminDashboard/room/page.tsx]: service missing

**Pull request closed without merging**